x-pack/plugins/observability_log_explorer/public/utils/breadcrumbs.tsx

```diff
@@ -9,33 +9,67 @@ import { EuiBreadcrumb } from '@elastic/eui';
 import type { ChromeStart } from '@kbn/core-chrome-browser';
 import type { ServerlessPluginStart } from '@kbn/serverless/public';
 import { useEffect } from 'react';
-import { logExplorerAppTitle } from '../../common/translations';
+import { useLinkProps } from '@kbn/observability-shared-plugin/public';
+import {
+  logExplorerAppTitle,
+  logsAppTitle,
+  observabilityAppTitle,
+} from '../../common/translations';
+import {
+  OBSERVABILITY_LOG_EXPLORER_APP_ID,
+  OBSERVABILITY_OVERVIEW_APP_ID,
+  OBSERVABILITY_LOGS_APP_ID,
+} from '../../common/constants';
 
 export const useBreadcrumbs = (
   breadcrumbs: EuiBreadcrumb[],
   chromeService: ChromeStart,
   serverlessService?: ServerlessPluginStart
 ) => {
+  const observabilityLinkProps = useLinkProps({ app: OBSERVABILITY_OVERVIEW_APP_ID });
+  const logsLinkProps = useLinkProps({ app: OBSERVABILITY_LOGS_APP_ID });
+  const logExplorerLinkProps = useLinkProps({ app: OBSERVABILITY_LOG_EXPLORER_APP_ID });
+
   useEffect(() => {
-    setBreadcrumbs(breadcrumbs, chromeService, serverlessService);
-  }, [breadcrumbs, chromeService, serverlessService]);
+    setBreadcrumbs(
+      serverlessService
+        ? breadcrumbs
+        : [
+            {
+              text: observabilityAppTitle,
+              ...observabilityLinkProps,
+            },
+            {
+              text: logsAppTitle,
+              ...logsLinkProps,
+            },
+            {
+              text: logExplorerAppTitle,
+              ...logExplorerLinkProps,
+            },
```

Comment:
**note:** This custom hook is also [used by the Dataset quality page](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_log_explorer/public/routes/main/dataset_quality_route.tsx#L23-L29) to set its breadcrumbs. Applying the log explorer breadcrumbs here means it will also appear for that page.
To keep it reusable, let's keep here only the `Observability` and `Logs` breadcrumbs, while we can pass the Log explorer title from the consumer component for this page, wdyt?

Reply:
Agree with you, and actually looking better at the design the Dataset quality should be a sub breadcrumb of the log explorer, so your implementation is totally fine 👌